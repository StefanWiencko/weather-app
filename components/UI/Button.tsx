import { Platform, Pressable, StyleSheet, Text, View } from "react-native";
import React from "react";
import { Colors } from "@/constants/colors";
type Props = {
  children: React.ReactNode;
  onPress: (...args: any[]) => void;
  style?: Record<string, number | string>;
};

const Button = ({ children, onPress, style }: Props) => {
  return (
    <View style={[styles.rootContainer, style]}>
      <Pressable
        android_ripple={{ color: Colors.accent200 }}
        onPress={onPress}
        style={styles.button}
      >
        <Text style={styles.buttonText}>{children}</Text>
      </Pressable>
    </View>
  );
};

export default Button;

const styles = StyleSheet.create({
  rootContainer: {
    overflow: Platform.OS === "android" ? "hidden" : "visible",
    borderRadius: 4,
  },
  button: {
    borderRadius: 4,
    padding: 8,
    backgroundColor: Colors.primary700,
  },
  buttonText: {
    color: "white",
    textAlign: "center",
  },
  pressed: {
    opacity: 0.75,
    backgroundColor: Colors.accent200,
    borderRadius: 4,
  },
});
