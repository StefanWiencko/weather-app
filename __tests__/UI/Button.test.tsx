import Button from "@/components/UI/Button";
import { render, fireEvent } from "@testing-library/react-native";

describe("Button", () => {
  it("displays text correctly", () => {
    const { getByText } = render(<Button onPress={() => {}}>Test</Button>);

    const button = getByText("Test");
    expect(button).toBeTruthy();
  });

  it("fires onPress fn", () => {
    const mockOnPress = jest.fn();
    const { getByText } = render(<Button onPress={mockOnPress}>Test</Button>);

    const button = getByText("Test");
    fireEvent.press(button);

    expect(mockOnPress).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const tree = render(<Button onPress={() => {}}>Test</Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
