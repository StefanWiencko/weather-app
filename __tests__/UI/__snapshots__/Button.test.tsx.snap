// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<View
  style={
    [
      {
        "borderRadius": 4,
        "overflow": "visible",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#6666ff",
        "borderRadius": 4,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "textAlign": "center",
        }
      }
    >
      Test
    </Text>
  </View>
</View>
`;
